Library/Homebrew/test/cmd/--prefix_spec.rb

```diff
@@ -13,7 +13,7 @@
       .and be_a_success
   end
 
-  it "prints the prefix for a Formula", :integration_test do
+  it "prints the prefix for a Formula", :integration_test, :needs_homebrew_core do
```

Comment:
This should already work without homebrew/core? The command is definitely working without 🤔

Reply:
It was the only other test that failed for me locally after untapping `homebrew/core`. I believe it couldn't load the formula. Feel free to update it if you can get it working.

```console
$ brew untap homebrew/core
Untapping homebrew/core...
Untapped 3 commands and 7053 formulae (7,402 files, 807.5MB).
$ brew tests --only=cmd/--prefix
Randomized with seed 37956
1 process for 1 spec, ~ 1 spec per process
...F.

Failures:

  1) Homebrew::Cmd::Prefix prints the prefix for a Formula
     Failure/Error:
       expect { brew_sh "--prefix", "wget" }
         .to output("#{ENV.fetch("HOMEBREW_PREFIX")}/opt/wget\n").to_stdout
         .and not_to_output.to_stderr
         .and be_a_success

          expected block to output "/usr/local/opt/wget\n" to stdout, but output nothing

       ...and:

             expected block to not output to stderr, but output "Error: No available formula with the name \"wget\".\n"

          ...and:

             expected #<Proc:0x000000011089c900 /usr/local/Homebrew/Library/Homebrew/test/cmd/--prefix_spec.rb:17> to be a success
       Diff for (output "/usr/local/opt/wget\n" to stdout):
       @@ -1,2 +1 @@
       -/usr/local/opt/wget

     # ./test/cmd/--prefix_spec.rb:17:in `block (2 levels) in <top (required)>'
     # ./test/support/helper/spec/shared_context/integration_test.rb:49:in `block (2 levels) in <top (required)>'



Took 11 seconds
Tests Failed
```

Reply:
This should be resolvable by adding a `test_formula` or similar which is written to the relevant location on disk or, as below, writing a mock API JSON to disk.

Reply:
Would that work for an integration test? I assumed that integration tests used the repo in it's normal state since they shell out to the `brew` executable so any test formula would need to be added to one of the non-test repos which seems less than ideal.

Reply:
> Would that work for an integration test?

Yes, other integration tests use non-core formulae.

Also, in this case: I'm pretty sure writing to `${HOMEBREW_CACHE}/api/formula_names.txt` is sufficient as `formula_path.sh` is the main logic here.